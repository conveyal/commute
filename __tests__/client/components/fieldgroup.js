/* global describe, expect, it, jest */

import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import React from 'react'

import FieldGroup from '../../../client/components/fieldgroup'

describe('Container > FieldGroup', () => {
  it('renders correctly as input', () => {
    const props = {
      label: 'Name',
      name: 'name',
      onChange: jest.fn(),
      placeholder: 'Enter name',
      type: 'text'
    }
    // mount component
    const tree = mount(
      <FieldGroup
        {...props}
        />
    )
    expect(mountToJson(tree)).toMatchSnapshot()
  })
})
