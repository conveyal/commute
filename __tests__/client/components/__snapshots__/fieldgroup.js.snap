exports[`Container > FieldGroup renders correctly as input 1`] = `
<FieldGroup
  label="Name"
  name="name"
  onChange={[Function]}
  placeholder="Enter name"
  type="text">
  <FormGroup
    bsClass="form-group"
    controlId="group-item-name">
    <div
      className="form-group">
      <ControlLabel
        bsClass="control-label"
        srOnly={false}>
        <label
          className="control-label"
          htmlFor="group-item-name">
          Name
        </label>
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        onChange={[Function]}
        placeholder="Enter name"
        type="text"
        value="">
        <input
          className="form-control"
          id="group-item-name"
          onChange={[Function]}
          placeholder="Enter name"
          type="text"
          value="" />
      </FormControl>
    </div>
  </FormGroup>
</FieldGroup>
`;
